Services/Component/classes/Setup/class.ilComponentDefinitionsStoredObjective.php

```diff
@@ -47,7 +47,9 @@ public function getPreconditions(Setup\Environment $environment) : array
     {
         return [
             new \ilDatabaseUpdatedObjective(),
-            new \ilDatabaseUpdateStepsExecutedObjective(new ilCOPageDBUpdateSteps())
+            new \ilDatabaseUpdateStepsExecutedObjective(new ilCOPageDBUpdateSteps()),
```

Comment:
@klees Could you please have a look at the changes in this file?

Reply:
Please add the `Objective` for the component repository explicitly too. Currently, `RESOURCE_COMPONENT_REPOSITORY` (attention, name change!) only exists by accident, as `RESOURCE_COMPONENT_FACTORY` depends on it. This could just change and this code would break. It's best too be super explicit here, IMO.